### PR TITLE
chore: add instructions about the `python modules error` to the troubleshoothing

### PR DESCRIPTION
**Pull request by LuizSDCit**
Add a solution in the troubleshooting for the python module access error.

Explain to the user that he must grant the artifact registry reader role to the appropriate Dataflow worker Service account.

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

File: docs/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -132,3 +133,61 @@ The **Service Account** must be declared as a job parameter.
...
+
+```console
+https://LOCATION-python.pkg.dev/ARTIFACT-REGISTRY-PROJECT-ID/python-modules/simple/
+ERROR: Could not find a version that satisfies the requirement apache-beam==2.30.0 (from versions: none)
+ERROR: No matching distribution found for apache-beam==2.30.0
+```
+
+**Cause:**
+
+The Dataflow [Worker Service Account](https://cloud.google.com/dataflow/docs/concepts/security-and-permissions#worker_service_account) is trying to access the [Artifact Registry](https://cloud.google.com/artifact-registry/docs/python/quickstart) to download the apache beam module but do not have the right permissions to access it.
```
Comment (amandakarina) [inline docs/TROUBLESHOOTING.md:152]: *Apache Beam*
```suggestion
The Dataflow [Worker Service Account](https://cloud.google.com/dataflow/docs/concepts/security-and-permissions#worker_service_account) is trying to access the [Artifact Registry](https://cloud.google.com/artifact-registry/docs/python/quickstart) to download the **Apache Beam module** but do not have the right permissions to access it.
```

File: docs/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -132,3 +133,61 @@ The **Service Account** must be declared as a job parameter.
...
+
+**Solution:**
+
+You must grant the role Artifact Registry Reader (`roles/artifactregistry.reader`) in the Artifact Registry Repository the has the python modules to the Dataflow Worker Service Account.
```
Comment (amandakarina) [inline docs/TROUBLESHOOTING.md:156]:
```suggestion
You must grant the role Artifact Registry Reader (`roles/artifactregistry.reader`) in the Artifact Registry Repository where the python modules are to the Dataflow Worker Service Account.
```

**Review by amandakarina — APPROVED** [review]
LGTM

File: docs/TROUBLESHOOTING.md (hunk 2 above)
Comment (daniel-cit) [inline docs/TROUBLESHOOTING.md:156]:
```suggestion
You must grant the role Artifact Registry Reader (`roles/artifactregistry.reader`) in the Artifact Registry Repository, that host the Python modules, to the Dataflow Worker Service Account.
```

File: docs/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -132,3 +133,61 @@ The **Service Account** must be declared as a job parameter.
...
+--location=${location}
+```
+
+Using terraform:
+
+```hcl
+resource "google_artifact_registry_repository_iam_member" "python_reader" {
+  provider = google-beta
+
+  project    = <ARTIFACT-REGISTRY-PROJECT-ID>
```
Comment (erlanderlo) [inline docs/TROUBLESHOOTING.md:187]: Although I understand the example, consider use the normal terraform substitution methods as an example such as defining some locals and then referencing them as `local.project_id` or else use an actual example string such as `project = "CONFIDENTIAL-DATA-PROJECT-ID"`
`serviceAccount:sa-dataflow-controller-reid@CONFIDENTIAL-DATA-PROJECT-ID.iam.gserviceaccount.com`

File: docs/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -132,3 +133,61 @@ The **Service Account** must be declared as a job parameter.
 - **Terraform**: Use the input `service_account_email` from the [Dataflow Flex Job Module](../modules/dataflow-flex-job/README.md#inputs).
 
 For more details about Dataflow staging files see [Resource usage and management](https://cloud.google.com/dataflow/docs/guides/deploying-a-pipeline#resource-usage-and-management) documentation.
+
+### No matching distribution found for apache-beam==2.30.0
+
+An user deploys a new Dataflow job in the console and the job creation fails.
```
Comment (erlanderlo) [inline docs/TROUBLESHOOTING.md:139]: consider "After deploying a new Dataflow job in the console, the job creation fails."

**Review by erlanderlo — APPROVED** [review]
Thanks. some minor things, but generally will be helpful to add!  LGTM

File: docs/TROUBLESHOOTING.md (hunk 3 above)
Comment (LuizSDCit) [reply docs/TROUBLESHOOTING.md]: @erlanderlo I made the change. Could you PTAL?

**Pull request merged**